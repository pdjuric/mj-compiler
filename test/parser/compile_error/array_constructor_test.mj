program function_call_test

int i;
char c;

{
    void main() int arr[];  {

        arr = new int['a'];
        
        c = 'c';
        arr = new arr[c];

        arr = new int[false];

        // array access testing
        i = c[5];
        i = i[2];
   }
}