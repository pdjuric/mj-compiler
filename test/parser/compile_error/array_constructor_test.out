INFO  14:01:59,039 - line   9 - Access to local variable arr [Var arr: Arr of int, 0, 1 ].
ERROR 14:01:59,039 - line   9 - Type of array index expression must be int.
INFO  14:01:59,040 - line  11 - Access to global variable c [Var c: char, 1, 0 ].
INFO  14:01:59,040 - line  12 - Access to local variable arr [Var arr: Arr of int, 0, 1 ].
ERROR 14:01:59,040 - line  12 - arr is not a type
INFO  14:01:59,040 - line  12 - Access to global variable c [Var c: char, 1, 0 ].
ERROR 14:01:59,040 - line  12 - Type of array index expression must be int.
INFO  14:01:59,040 - line  14 - Access to local variable arr [Var arr: Arr of int, 0, 1 ].
ERROR 14:01:59,041 - line  14 - Type of array index expression must be int.
INFO  14:01:59,041 - line  17 - Access to global variable i [Var i: int, 0, 0 ].
INFO  14:01:59,041 - line  17 - Access to global variable c [Var c: char, 1, 0 ].
ERROR 14:01:59,041 - line  17 - c is not an array
INFO  14:01:59,041 - line  18 - Access to global variable i [Var i: int, 0, 0 ].
INFO  14:01:59,042 - line  18 - Access to global variable i [Var i: int, 0, 0 ].
ERROR 14:01:59,042 - line  18 - i is not an array